### PR TITLE
Create a setting to allow using the new kql extension, or the old csl extension

### PR DESCRIPTION
**Pull request by anladwig**
Create a setting to allow using the new kql extension, or the old csl extension, with kql being the default.

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,5 +27,8 @@ If your username is in the form of UPN (user@contoso.com), you can hit the commo
 - Place table fields on new lines - When CSL table files are written to a file target, each field is placed on a new line. This can make it easier to diff changes in a pull request.
 - Generated ".create-merge table" commands instead of ".create table" - When CSL table files are written to a file target, the command will be ".create-merge table" if this is checked. If it is unchecked, the command will be ".create table".
 
+### Files
+- Use legacy `.csl` file extensions - Kusto originally used files with `.csl` file extensions. This extension has since been deprecated and replaced with the `.kql` file extension. This setting tells synckusto which file extension should be used for reading _and_ writing all files. The default state is unchecked, which means synckusto will consume and emit files with the `.kql` file extension.
```
Comment (benmartens) [inline README.md:31]: I think I understand the motivation here, but making kql the default means that everyone who already has a big tree of csl files will get "broken" by this. All of their files will change when they upgrade to this. So this feature is fine but we need to keep csl as the default.

**Comment by benmartens** [comment]
The build number gets displayed in the top right corner and the super fancy way we do this is in MainForm.Designer.cs: https://github.com/microsoft/synckusto/blob/3a29a3670c0ab053ec96f246ec6ea5e537b4c5ef/SyncKusto/MainForm.Designer.cs#L153

Please updated this as part of your change.

File: README.md (hunk 1 above)
Comment (anladwig) [reply README.md:31]: I think that's fair, and its as simple a changing the default to be checked and leaving the rest of the setting wording to be the same.

Comment (benmartens) [inline README.md:31]: Thanks for the change, but I think the readme needs to be updated too?

"The default state is checked, which means synckusto will consume and emit files with the .csl file extension."

**Review by benmartens — APPROVED** [review]

**Pull request merged**